watcher/lib/influxdb/influxdb.go

```diff
@@ -69,38 +83,100 @@ func (c *Client) Setup() error {
 	for _, query := range queries {
 		log.Infof("%v", query)
 
-		response, err := c.PostForm(c.Endpoint("query"), url.Values{"q": []string{query}})
+		err := c.execQuery(query)
 		if err != nil {
+			if trace.IsAlreadyExists(ConvertInfluxDBError(err)) {
+				log.Info("Retention policy already exists with different attributes.")
+				continue
+			}
 			return trace.Wrap(err)
 		}
-
-		log.Infof("%v %v %v", response.Code(), response.Headers(), string(response.Bytes()))
 	}
 	return nil
 }
 
-// CreateRollup creates a new rollup query in the database
+// CreateRollup creates a rollup query in the database
```

Comment:
Creates a new query or updates an existing one? What happens if the query for this rollup has already been created?